src/Symfony/Component/HttpFoundation/RequestStack.php

```diff
@@ -26,6 +26,16 @@ class RequestStack
      */
     private array $requests = [];
 
+    /**
+     * @param Request[] $requests
+     */
+    public function __construct(array $requests = [])
```

Comment:
```suggestion
    public function __construct(Request|array $requests = [])
```
does that make sense?

Reply:
We require atleast an iterable else we would need to change the logic to something like this:

```
__construct(Request|array $requests = [])
{
    $requests = \is_array($requests) ? $requests : [$requests];
    
    foreach ($requests as $request) {
         $this->push($request);
    }
}
```

Not sure if worth. Personally not a fan of Union types. But if others agree I can implement the allowness of a single request object.

Reply:
alternative would be factory methods:

```php
RequestStack::fromRequest($request);
```

```php
public static function fromRequest(Request $request)
{
    return new self([$request]);
}
```